### ISSUE TITLE
Write an howto install fresh 15.04

### ISSUE THREAD

**Issue by domak** [advancingu/XPS13Linux#18]
As mention in #8, upgrading from 14.04 is not possible. 
I read in dell forum that one guy bricks its laptop while attempting to install a fresh 15.04 (even the recovery process was impossible).
So, if somebody who was successful to install 15.04 could write a github wiki page to put some trick I will fill more comfortable (how to save the current settings to optionally recover, BIOS or UEFI, etc).
Thanks

**Comment by tgalopin** [comment]
I will receive my XPS in a few days, I'll write an article about my experiments. According to https://github.com/mpalourdio/xps13 (config 9), the 3.19.0 kernel works fine with some drivers, so I plan to try to do this:
1. See what is currently not working ;
2. Try to install the 3.19.0 kernel on the stock Ubuntu and see if I can make things work ;
3. I would like to install elementaryOS (derivative of Ubuntu 14.04) instead of classic Ubuntu, so if the kernel 3.19.0 works well, I'll also try to install elementaryOS from scratch (with http://www.dell.com/support/article/us/en/04/SLN297060/en if needed) ;
4. If I can't make Ubuntu 14.04 work, I'll install Ubuntu 15.04 (also with http://www.dell.com/support/article/us/en/04/SLN297060/en if needed) ;

My plan is to find a fully working configuration out of the various tests by people here and share it in a single article.

**Comment by antonmry** [comment]
I wrote something similar for Fedora 21, it's [here](https://gist.github.com/antonmry/eb10405cbc0787fa3aaf)

**Comment by cboettig** [comment]
I did a fresh install to ubuntu 15.04 from usb key on my xps 13 dev edition (after upgrading failed).  I had to install the proprietary wireless driver from the usb image (as described [here](http://askubuntu.com/questions/553615/), for some reason just adding the iso as the source under Software Sources gui and clicking 'install proprietary drivers' didn't work).  I also had to install Bluetooth drivers (as in #1).  I get about 5.5 hrs on the battery (display dimmed, wireless & bluetooth on, minimal cpu tasks) whereas [config 11](https://github.com/mpalourdio/xps13#config-11-by-rpbaptist) reports almost twice that -- maybe that's all due to the patched 4.0 kernel?.  Sometimes see flickering when connecting to an external 4K display; and occasionally get repeated keys (sometimes worse on my bluetooth K810 keyboard but also on the built-in one). trackpad also seems to be active while typing (leading to accidental presses) despite having 'disable while typing' checked.

**Comment by mcmillhj** [comment]
I did a fresh install of 15.04 from a USB key and did not need to install any drivers at all. Both wireless and bluetooth were working out of the box. I did have to disable UEFI boot mode though.

**Comment by anaran** [comment]
@mcmillhj I wonder what wireless hardware and driver you have then.

Here is mine after choosing Additional Drivers:

```
$ sudo lshw -C network
[sudo] password for user: 
  *-network               
       description: Wireless interface
       product: BCM4352 802.11ac Wireless Network Adapter
       vendor: Broadcom Corporation
       physical id: 0
       bus info: pci@0000:02:00.0
       logical name: wlan0
       version: 03
       serial: ...
       width: 64 bits
       clock: 33MHz
       capabilities: pm msi pciexpress bus_master cap_list ethernet physical wireless
       configuration: broadcast=yes driver=wl0 driverversion=6.30.223.248 (r487574) ip=... latency=0 multicast=yes wireless=IEEE 802.11abg
       resources: irq:19 memory:f7200000-f7207fff memory:f7000000-f71fffff
user@host-xps-13-9343:/var/log$ 
```

**Comment by mcmillhj** [comment]
@anaran I see no differences between your output and my own, but I did not install any additional drivers myself. I did do the typical `apt-get update && apt-get upgrade` after the install completed, this may have installed a driver that I didn't notice.

```
[hunter@eros: ~]$ sudo lshw -C network
[sudo] password for hunter: 
  *-network               
       description: Wireless interface
       product: BCM4352 802.11ac Wireless Network Adapter
       vendor: Broadcom Corporation
       physical id: 0
       bus info: pci@0000:02:00.0
       logical name: wlan0
       version: 03
       serial: c4:8e:8f:f3:34:a9
       width: 64 bits
       clock: 33MHz
       capabilities: pm msi pciexpress bus_master cap_list ethernet physical wireless
       configuration: broadcast=yes driver=wl0 driverversion=6.30.223.248 (r487574) ip=192.168.1.17 latency=0 multicast=yes wireless=IEEE 802.11abg
       resources: irq:19 memory:f7200000-f7207fff memory:f7000000-f71fffff
```

**Comment by domak** [comment]
@mcmillhj Did you install ubuntu with an ethernet cable?

**Comment by mcmillhj** [comment]
@domak Nope, I returned the USB3 to Ethernet adapter Dell marked as compatible with the XPS13 because it seemed to only work for the Ubuntu version that came with it. The Debian netinst image I was trying to install couldn't find the driver for it.

**Comment by tgalopin** [comment]
I installed the stock Ubuntu and everything seems to work fine. I'll investigate a bit more.

**Comment by domak** [comment]
@mcmillhj thanks. It's weird. so many people here get problems with broadcom wifi and bluetooth...
And why did you have to disable UEFI boot mode?

**Comment by cboettig** [comment]
I've also had to disable UEFI mode, perhaps because I have encrypted the
drive?  Device does not find any bootable medium when booting in UEFI.

**Comment by tgalopin** [comment]
I am not able to install properly Ubuntu 15.04, it installs but i cannot boot. @mcmillhj you are in Legacy mode right?

**Comment by mcmillhj** [comment]
@cboettig Same here. 

@domak after my install I got a black screen with a message that said 

```
no bootable devices found
```

I rebooted the machine and edited the BIOS to start in Legacy Boot mode and the machine booted. I am not entirely sure why I needed to make this change either. 

@cboettig Try editing your BIOS to start in Legacy Boot mode

**Comment by macghriogair** [comment]
@cboettig Same happened to me after a fresh install of Ubuntu 15.04 with whole disk encryption.
However, it was possible to add manually a boot option selecting /boot/EFI/shimx64.EFI from the F2 BIOS menu. After that, Ubuntu booted and most oft the previuos issues (e.g. trackpad) have gone so far.

Still I observe an occasionally flickering Ubuntu logo upon shutdown and hotkeys stopping to work every now and then...

**Comment by mcmillhj** [comment]
@macghriogair  I also see that every time I shutdown, I assume you are still having the suspend/resume issue as well?

**Comment by macghriogair** [comment]
@mcmillhj No, I could not observe suspend/resume issues so far, although did no excessive testing yet.

Concerning the hotkeys, I suspect it has to do with using Guake terminal. Every time it's visibility get's toggled via F12, the hotkeys, inlcuding Fn-keys and power button stop responding until I activate any other window, e.g. from the launcher. After all, this might not be a specific issue with the Dell XPS but with Guake or compiz.

**Comment by tgalopin** [comment]
I managed to install a fresh 15.04. Here is what I did:
- make a classic USB bootable stick with Ubuntu ;
- boot in Legacy Mode (Secure boot: OFF) on the stick (I had a boot problem about an unknow key in the boot config file: I had to type "live" to boot on the key)
- install Ubuntu 15.04 using the live USB ;

I had a Wifi problem: the Broadcom driver wasn't installed. I installed it using the same stick:
- boot on Ubuntu
- plug the stick in, go in it with the file explorer ;
- go to `pool/main/d/dkms` and install the .deb file (it's dkms, required for the broadcom driver) ;
- go to `pool/restricted/b/broadcom` and install the .deb file (the driver itself) ;
  
  Once done, the Wifi was accessible and I was able to update/upgrade the distribution. I still have two problems:
- the suspend mode does not work (it suspend properly, but the resume fail) ;
- I have repeated keys sometimes ;

I can deal with the suspend problem, but the repeated keys are pretty annoying :) . I'm trying to see how to solve these issues.

**Comment by tgalopin** [comment]
Okay, so it seems the suspend issue was a problem of display: I added the command `xrefresh -black` to the script just after the resume and it seems to work. I'll see in the long run.

**Comment by allefeld** [comment]
The UEFI booting problem might have to do with the BIOS expecting a file

```
boot/bootx64.efi
```

in the EFI partition, but the OS does install something else. Debian e.g. installs the EFI bootloader to

```
debian/grubx64.efi
```

A workaround is to mount the EFI partition (as vfat) using a rescue disk, and renaming the file/path.

**Comment by hg8** [comment]
Hello, I have wrote [a step-by-step guide for installing Ubuntu 15.04 on the XPS 13](http://hgdev.co/installing-ubuntu-15-04-on-the-xps-13-9343-2015-a-complete-guide/).
Hope it will be helpful.

**Comment by cboettig** [comment]
@hg8  very nice :trophy:

**Comment by hg8** [comment]
@cboettig Thanks :)

**Comment by PedramVeisi** [comment]
@hg8 This is perfect. Thanks.
Just one point, this command:
`sudo cp powersaverXPS13Vivid cd /etc/pm/power.d/`
has an extra cd.

**Comment by hg8** [comment]
Indeed, I have corrected it thanks !
